GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml

```diff
@@ -9,16 +9,19 @@
     <!-- Read and Write log files to storage -->
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <!-- Allow foreground service -->
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
 
     <application>
         <!-- SERVICES-->
         <service android:name="com.google.sharedlibrary.service.GpsDataCaptureService"
             android:exported="true"
             android:enabled="true">
             <intent-filter>
-                <action android:name="com.google.sharedlibrary.service.GpsDataCaptureService.STARTCAPTURE"/>
-                <action android:name="com.google.sharedlibrary.service.GpsDataCaptureService.STOPCAPTURE"/>
+                <action android:name="com.google.gpsdatacapturer.STOP_CAPTURE"/>
```

Comment:
In a future patch, is it possible to move handling of intent with action com.google.gpsdatacapturer.STOP_CAPTURE to activity?

Reply:
I don't feel we can move it to activity,  as we can only start the activity once, right?

Reply:
https://developer.android.com/guide/topics/manifest/activity-element#lmode : Can you try making the activity "singleTask" / "singleInstance"